test/e2e_node/swap_test.go

```diff
@@ -25,126 +25,321 @@ import (
+					},
+				}
+
+				return pod
+			}
+
+			getSwapSizeByPercentage := func(swapPercentage float64) *resource.Quantity {
+				return resource.NewQuantity(int64(float64(swapCapacity.Value())*swapPercentage), resource.BinarySI)
+			}
+
+			getPageAlignedQuantities := func(quantity *resource.Quantity) (alignedDownwards, alignedUpwards *resource.Quantity) {
+				alignedDownwards = cloneQuantity(*quantity)
+				alignedUpwards = cloneQuantity(*quantity)
+
+				alignedDownwards.Sub(resource.MustParse("4Ki"))
+				alignedUpwards.Add(resource.MustParse("4Ki"))
+
+				return
+			}
+
+			ginkgo.It("should be able over-commit the node memory up to the auto-calculated swap limit", func() {
+				config, err := kubeletconfig2.GetCurrentKubeletConfigFromFile()
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
+				gomega.Expect(config.MemorySwap.SwapBehavior).To(gomega.Equal(types.LimitedSwap), "swap behavior is not LimitedSwap")
+
+				const swapPercentageToUse = 0.2
+				//var err error
+
+				// Since cgroup sizes are page aligned, make sure we're within a page size of the expected value
+				swapToUse := getSwapSizeByPercentage(swapPercentageToUse)
+				minSwapSizeToUse, maxSwapSizeToUse := getPageAlignedQuantities(swapToUse)
 
-		if !isSwapFeatureGateEnabled() || !isCgroupV2 || (isLimitedSwap && (qosClass != v1.PodQOSBurstable || memoryRequestEqualLimit)) {
-			ginkgo.By(fmt.Sprintf("Expecting no swap. feature gate on? %t isCgroupV2? %t is QoS burstable? %t", isSwapFeatureGateEnabled(), isCgroupV2, qosClass == v1.PodQOSBurstable))
-			expectNoSwap(f, pod, isCgroupV2)
-			return
-		}
+				stressSize := getStressSize(swapToUse)
+				stressPod := getStressPod(stressSize.Value(), swapPercentageToUse)
 
-		if !isLimitedSwap {
-			ginkgo.By("expecting unlimited swap")
-			expectUnlimitedSwap(f, pod, isCgroupV2)
-			return
-		}
+				ginkgo.By(fmt.Sprintf("creating a stress pod with stress size %d and swap percentage %g which is of size %d", stressSize.Value(), swapPercentageToUse, swapToUse.Value()))
+				stressPod = runPodAndWaitUntilScheduled(f, stressPod)
 
-		ginkgo.By("expecting limited swap")
-		expectedSwapLimit := calcSwapForBurstablePod(f, pod)
-		expectLimitedSwap(f, pod, expectedSwapLimit)
-	},
-		ginkgo.Entry("QOS Best-effort", v1.PodQOSBestEffort, false),
-		ginkgo.Entry("QOS Burstable", v1.PodQOSBurstable, false),
-		ginkgo.Entry("QOS Burstable with memory request equals to limit", v1.PodQOSBurstable, true),
-		ginkgo.Entry("QOS Guaranteed", v1.PodQOSGuaranteed, false),
-	)
-})
+				swapLimit, err := readCgroupFileFromHost(stressPod, cgroupV2SwapLimitFile)
+				gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
-// Note that memoryRequestEqualLimit is effective only when qosClass is PodQOSBestEffort.
-func getSwapTestPod(f *framework.Framework, qosClass v1.PodQOSClass, memoryRequestEqualLimit bool) *v1.Pod {
-	podMemoryAmount := resource.MustParse("128Mi")
+				gomega.Expect(swapLimit).ToNot(gomega.Equal("max"), "swap limit is not set")
```

Comment:
The test fails with 

```
  [FAILED] swap limit is not set
  Expected
      <string>: max
  not to equal
      <string>: max
  In [It] at: test/e2e_node/swap_test.go:244 @ 10/23/23 17:29:53.382
```
because busybox-container the value of `memory.swap.max` is set to `max` while the swap limit only set for the container `stress-container`